### PR TITLE
Improving priority queues with max-heap

### PR DESCRIPTION
**Pull request by PuneetTri**
Improving time complexity of Priority Queues with help of Max-heap. Insertion time complexity, O(N) -> O(log(N))

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new Java files are placed inside an existing directory.
- [x] All filenames are in all uppercase characters with no spaces or dashes.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: src/main/java/com/thealgorithms/datastructures/queues/PriorityQueues.java [hunk 1]
```diff
@@ -122,6 +181,7 @@ public static void main(String[] args) {
 
         for (int i = 3; i >= 0; i--) {
             System.out.print(myQueue.remove() + " "); // will print the queue in reverse order [10, 5, 3, 2]
+
```
Comment (debasishbsws) [inline src/main/java/com/thealgorithms/datastructures/queues/PriorityQueues.java:184]:
```suggestion
```
remove it

**Review by debasishbsws — CHANGES_REQUESTED** [review]

**Comment by debasishbsws** [comment]
Please add some JUnit Tests for this code
https://github.com/TheAlgorithms/Java/tree/master/src/test/java/com/thealgorithms/datastructures
You can refer to this

File: src/main/java/com/thealgorithms/datastructures/queues/PriorityQueues.java [hunk 2]
```diff
@@ -30,37 +32,94 @@ class PriorityQueue {
      * @param size Size of the queue
      */
     public PriorityQueue(int size) {
-        maxSize = size;
-        queueArray = new int[size];
+        /* If capacity is not defined, default size of 11 would be used
```
Comment (siriak) [inline src/main/java/com/thealgorithms/datastructures/queues/PriorityQueues.java:35]: Where is this 11?

File: src/main/java/com/thealgorithms/datastructures/queues/PriorityQueues.java [hunk 3]
```diff
@@ -8,6 +8,8 @@
  * most important elements are placed at the front/on the top. In this example I
  * give numbers that are bigger, a higher priority. Queues in theory have no
  * fixed size but when using an array implementation it does.
+ * 
+ *Addtional contibutions made by: PuneetTri(https://github.com/PuneetTri)
```
Comment (siriak) [inline src/main/java/com/thealgorithms/datastructures/queues/PriorityQueues.java:12]: Typo

**Review by siriak — CHANGES_REQUESTED** [review]
Please also add some JUnit tests (see https://github.com/TheAlgorithms/Java/tree/master/src/test/java/com/thealgorithms)

File: src/main/java/com/thealgorithms/datastructures/queues/PriorityQueues.java (hunk 3 above)
Comment (PuneetTri) [reply src/main/java/com/thealgorithms/datastructures/queues/PriorityQueues.java]: Thank you, fixed it!

File: src/main/java/com/thealgorithms/datastructures/queues/PriorityQueues.java (hunk 2 above)
Comment (PuneetTri) [reply src/main/java/com/thealgorithms/datastructures/queues/PriorityQueues.java:35]: I was going to add a default constructor but I guess I forgot at that time, Now I've added it

**Comment by PuneetTri** [comment]
Sorry, I wasn't able to see your messages @siriak and @debasishbsws. I had my semester practical examination going on, so didn't have much time left off after studying. Sorry for the late reply.
I have added JUnit testing as per your wish inside the test folder, JUnit testing PR  is #3778

**Comment by PuneetTri** [comment]
Sorry for the request spams, didnt know you cannot ask two people to review at same time

**Review by debasishbsws — APPROVED** [review]

**Review by debasishbsws — COMMENTED** [review]
Resolve the build fail.

File: src/main/java/com/thealgorithms/datastructures/queues/PriorityQueues.java [hunk 4]
```diff
@@ -1,4 +1,5 @@
-package com.thealgorithms.datastructures.queues;
+// package com.thealgorithms.datastructures.queues;
```
Comment (debasishbsws) [inline src/main/java/com/thealgorithms/datastructures/queues/PriorityQueues.java:1]:
```suggestion
package com.thealgorithms.datastructures.queues;
```

**Comment by debasishbsws** [comment]
@siriak review. Test-case is already merge through #3778

**Review by debasishbsws — DISMISSED** [review]

**Comment by PuneetTri** [comment]
@siriak please approve the changes, been waiting for 6 days

**Comment by PuneetTri** [comment]
@yanglbme @siriak approval is still pending!

**Comment by PuneetTri** [comment]
@siriak @yanglbme this is ridiculous, I don't understand why you guys still haven't approved the changes. Did I do something wrong, if not why is it taking so long?

**Review by yanglbme — APPROVED** [review]

**Comment by yanglbme** [comment]
@siriak please review the change

**Review by siriak — APPROVED** [review]

**Pull request merged**